### DISCUSSION TITLE
How to require a payment on the Front End Submit form?

### DISCUSSION THREAD

**Discussion by lintonhale** [liedekef/events-made-easy#632] [Q&A]
Hi Franky!

What we'd like to do on the Front End Submit form is to require a payment for submitting the event listing. Is that possible?

On this page
https://www.e-dynamics.be/wordpress/frontend-submit/
I see the event property 
use_braintree
which I have added to the Default form format on this page:
https://learnsonomacounty.com/wp-admin/admin.php?page=eme-options&tab=emefs

I have tried both
#_PROP{use_braintree}{binary}
and 
#_PROP{use_braintree(*)}{binary}
but either way I just see a binary No/Yes dropdown menu under "Payment" field on our Frontend Submit page:
https://learnsonomacounty.com/list-your-class/
not the hoped for payment options.

I searched the document for "payment" and found that this says "When showing the payment form..."
https://www.e-dynamics.be/wordpress/payment-gateway-placeholders/
but how do you show the payment form?

---
Here's a picture of the Payments settings:
https://app.screencast.com/gyTprfqBf01sY
I'm fairly certain I've got the proper Merchant ID, Public, and Private Keys.

---
Here's the Frontend Submit Default form format code:

<div class="input">
		<label for="event_name">Event Name</label><br />
		#_FIELD{event_name}
		</div>
                <div class="input select">
                        <label for="event_category_ids">Select the Event Category</label><br />
			#_FIELD{event_category_ids}
                </div>
		<fieldset>
			<legend>Date & Time</legend>
			<fieldset id="event_start">
				<legend>Start</legend>
				<div class="input">
					<label for="event_start_date">Date</label>
					#_FIELD{event_start_date}
				</div>
				<div class="input">
					<label for="event_start_time">Time</label> 
					#_FIELD{event_start_time}
				</div>
			</fieldset>
			
			<fieldset id="event_end">
				<legend>End</legend>
				<div class="input">
					<label for="event_end_date">Date</label>
					#_FIELD{event_end_date}
				</div>
				<div class="input">
					<label for="event_end_time">Time</label> 
					#_FIELD{event_end_time}
				</div>
			</fieldset>
		</fieldset>

                <div class="input">
                        <label for="use_braintree}">Payment</label>
                        #_PROP{use_braintree (*)}{binary}
                </div>

            <div class="input">
                        <label for="short_description">Short Description (Max 120 characters)</label><br />
			#_CUSTOMFIELD{short_description}
                </div>

                <div class="input">
                        <label for="event_notes">Description</label><br />
			#REQ_FIELD{event_notes}
                </div>

                <div class="input">
                        <label for="price">Cost</label>
                        #_FIELD{price}
                </div>

                <div class="input">
                        <label for="event_contactperson_email_body">Contact Email</label><br />
			#REQ_FIELD{event_contactperson_email_body}
                </div>

                <div class="input">
                        <label for="upload_image">Upload Image (Please provide square image!)</label><br />
			#_CUSTOMFIELD{upload_image}
                </div>
                <div class="input">
                        <label for="event_url">Event Web Page</label><br />
			#_FIELD{event_url}
                </div>
                <h3>Location Information</h3>
                <div class="input">
                        <label for="location_name">Name of Venue</label>
                        #_FIELD{location_name}
                </div>

                <div class="input">
                        <label for="location_address1">Address</label>
                        #_FIELD{location_address1}
                </div>

                <div class="input">
                        <label for="location_city">City or Town</label>
                        #_FIELD{location_city}
                </div>

                <div class="map">
                        #_MAP
                </div>

**Comment by liedekef** [comment]
The payment property should be entered like this:
`#_PROP{use_braintree}{binary}`

But if you actually want people to pay for the event submission (as they do for bookings or memberships): that's currently not possible: this would require some extra development (extra event status, optional cleanup after X time when unpaid, handling the payment, maybe mail templates ...). Not impossible but this is the kind of thing I'm hoping to get a donation for :-)

**Comment by lintonhale** [comment]
Thanks Franky. Yes, we'd be interested in having your help with this and are happy to make a donation. I'll follow up via email and include the client in the conversation. Much appreciated!

**Reply by liedekef** [reply]
I'll be happy to work on that, feel free to contact me.

**Reply by liedekef** [reply]
The just released version has this (took some days though ... lot's of code reorg too, to improve general code reuse)

**Comment by lintonhale** [comment]
Great, thanks for letting me know. I apologize if you feel that we owe you
the $250. We asked how much, but did not ask you to proceed and did not
make an agreement to pay.

My client does not yet have the income, so we found a free alternative.
When she makes use of the new features we will plan to make a donation.
Best wishes, Linton

On Thu, Aug 22, 2024 at 2:16 AM Franky Van Liedekerke <
***@***.***> wrote:

> The just released version has this (took some days though ... lot's of
> code reorg too, to improve general code reuse)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/liedekef/events-made-easy/discussions/632#discussioncomment-10416997>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZIQ4VPG3IVBRNHQNT27DTZSWUAFAVCNFSM6AAAAABMPHSWXGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBRGY4TSNY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 

Linton Hale

Web Developer, Internet Coach, Musician
707-206-5442 ***@***.***

<https://webwatchdawg.com/>
Terms of Service <https://webwatchdawg.com/terms-of-service/> | Privacy
Policy <https://webwatchdawg.com/privacy-policy/>

**Reply by liedekef** [reply]
If I wanted the money, I would not have implemented it :-) I'm living on donations (well ... much less than before though), so I hope it will help people.

**Comment by lintonhale** [comment]
I'm glad to hear that your work is being supported. Thanks for your help!

On Thu, Aug 22, 2024 at 8:37 AM Franky Van Liedekerke <
***@***.***> wrote:

> If I wanted the money, I would not have implemented it :-) I'm living on
> donations (well ... much less than before though), so I hope it will help
> people.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/liedekef/events-made-easy/discussions/632#discussioncomment-10421058>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABZIQ4XEJ2EV3FJ7R5HMWKLZSYAUZAVCNFSM6AAAAABMPHSWXGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBSGEYDKOA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 

Linton Hale

Web Developer, Internet Coach, Musician
707-206-5442 ***@***.***

<https://webwatchdawg.com/>
Terms of Service <https://webwatchdawg.com/terms-of-service/> | Privacy
Policy <https://webwatchdawg.com/privacy-policy/>

**Reply by liedekef** [reply]
well, since I moved to github I'm getting a lot less donations, but such is life ...
